modules/ace/commands.ts

```diff
@@ -8,16 +8,49 @@
  */
 
 import { BaseCommand as AceBaseCommand, ListCommand as AceListCommand } from '@adonisjs/ace'
+import { slash } from '@poppinss/utils'
 
 import { Kernel } from './kernel.js'
 import type { ApplicationService } from '../../src/types.js'
 import type { CommandOptions, ParsedOutput, UIPrimitives } from '../../types/ace.js'
 
+/**
+ * Wrapper around the stub generation logic.
+ * Allow commands to easily generate files from given stubs
+ */
+class StubGenerator {
+  #command: BaseCommand | ListCommand
+  #flags: Record<string, any>
+
+  constructor(command: BaseCommand, flags: Record<string, any>) {
+    this.#command = command
+    this.#flags = flags
+  }
+
+  async generate(stubsRoot: string, stubPath: string, stubState: Record<string, any>) {
+    const stub = await this.#command.app.stubs.build(stubPath, { source: stubsRoot })
+    const output = await stub.generate(Object.assign({ flags: this.#flags }, stubState))
+
+    const entityFileName = slash(this.#command.app.relativePath(output.destination))
+    const result = { ...output, relativeFileName: entityFileName }
+
+    if (output.status === 'skipped') {
+      this.#command.logger.action(`create ${entityFileName}`).skipped(output.skipReason)
+      return result
+    }
+
+    this.#command.logger.action(`create ${entityFileName}`).succeeded()
+    return result
+  }
+}
+
```

Comment:
Do we need this class? Can't we just inline all the `generate` method code inside the `makeUsingStub` method?

Reply:
This is to avoid duplicating the code in the ListCommand class that implements BaseCommand as I explain in my comment above
As it is the case for the `terminate` function for example

Reply:
Ohhh, I see. I think, we will run into this issue more often as we try to expand the capabilities of the base command. 

Should we fix it right now maybe using mixins?